Documentation/internal/README.md

```diff
@@ -1,5 +1,160 @@
+process executable metadata into `Process` struct, scanning for:
+
+* loadProcessInformation - general process information (/proc/\*/comm, /proc/\*/stat)
+* parseDebugFrame - parses ELF `.debug_frame` and `.debug_info` sections
+* obtainGoSymbols - parses ELF `.gosymtab`, `.gopclntab` and `.text` sections
+* parseDebugLineInfo - parses ELF `.debug_line` section
+* loadTypeMap - parses DWARF types map
+
+To finish threads list updated, go compiler version scanned, `G` struct offset set, current `g` loaded,
+location of `runtime.startpanic` found and to the most end the breakpoints inited.
+
+## Memory IO
+
+To be written...
+
+## G parser
+
+To be written...
+
+## Breakpoints implementation
+
+To be written...
+
+## `Process.new`
+
+Initialization of the Process struct consists of setting up `ptrace` environment, debugging process
+pid, os/arch details and some struct for receiving future info from the victim.
+
+`ptrace` initialization (`handlePtraceFuncs`) uses `runtime.LockOSThread()` call to lock current
+goroutine to the OS thread due properly setup `ptrace` env. This is due to the fact that ptrace(2) expects
+all commands after PTRACE_ATTACH to come from the same thread.
+
+## `ptrace` syscall
+
+_The ptrace system call is crucial to the working of debugger programs like gdb._
+
+ptrace() is a system call that enables one process to control the execution of another. 
+It also enables a process to change the core image of another process. The traced process behaves normally 
+until a signal is caught. When that occurs the process enters stopped state and informs the tracing 
+process by a wait() call. Then tracing process decides how the traced process should respond. 
+The only exception is SIGKILL which surely kills the process.
+
+The traced process may also enter the stopped state in response to some specific events during 
+its course of execution. This happens only if the tracing process has set any event flags in 
+the context of the traced process. The tracing process can even kill the traced one by setting 
+the exit code of the traced process. After tracing, the tracer process may kill the traced one 
+or leave to continue with its execution.
+
+_Note: Ptrace() is highly dependent on the architecture of the underlying hardware. 
+Applications using ptrace are not easily portable across different architectures and implementations._
+
+Link:
+
+* [ptrace(2) - Linux man page](https://linux.die.net/man/2/ptrace)
+* [Process Tracing Using Ptrace](http://tldp.org/LDP/LG/issue81/sandeep.html)
+* [ptrace - WikiPedia](https://en.wikipedia.org/wiki/Ptrace)
+
+## `ptrace` binding
+
+`ptrace` implementation uses 1 separate goroutine per debugging process (instance of the `ptrace`). To
+guarantee safe `ptrace` environment `runtime.LockOSThread` is used. Commands received via Go channel.
+
+# `dwarf` 
+
+`dwarf` stands for DWARF Debugging Information Format. The 6l and 8l linkers emit DWARF3 debugging information 
```

Comment:
6l and 8l don't exist anymore, I'm not even sure we support any version of go where 6l and 8l still exist. The linker is just called `link` now